### ISSUE TITLE
AutoMocker does not expose non-generic overloads

### ISSUE THREAD

**Issue by Keboo** [tkellogg/Moq.AutoMocker#25]
I am opening this issue to see if this is something that is desirable (I am happy to submit PRs if it is).

On several of my projects that use various DI containers, I tend to write a class that bridges between the DI container and AutoMocker to have the DI container automatically fill requests with mocks from AutoMocker. 
The issue is that the DI containers will pass the service type as a System.Type rather than calling a generic method. I tend to write extension methods on AutoMocker to expose non-generic overloads for  the methods Get, GetMock, and CreateInstance.

Things like this:

``` C#
public static object Get(this AutoMocker mocker, Type serviceType)
{
    return typeof(AutoMocker).GetMethod(nameof(AutoMocker.Get))
            .MakeGenericMethod(serviceType)
            .Invoke(mocker, new object[0]);
}
```

Is this something that would be considered?

**Comment by Keboo** [comment]
Fixed on fork https://github.com/adamhewitt627/Moq.AutoMocker

**Comment by adamhewitt627** [comment]
Fixed in #27.
